### PR TITLE
Renames `Frame`->`Adornment`; changes `Frame` to have a `Border` subclass

### PR DESCRIPTION
**Pull request by tig**
See this discussion: https://github.com/gui-cs/Terminal.Gui/issues/2994#issuecomment-1887861731

- [x] Create `Margin` subclass of `Frame`
- [x] Create `Border` subclass of `Frame`
- [x] Create `Padding` subclass of `Frame`
- [x] Improve robustness of Adornment creation
- [x] `Border` line style inherits from `SuperView` (if not set explicitly).
- [x] Update/tweak `Frames` scenario
- [x] `Frame`->`Adornment`
- [x] `Frames` scenario etc... rename 
- [x] More/better `Adornment` unit tests

### CODE REVIEW

**Comment by tig** [comment]
@BDisp take a look! Just a WIP, but...

File: Terminal.Gui/View/ViewDrawing.cs [hunk 1]
```diff
@@ -454,7 +454,7 @@ public virtual void OnDrawContent (Rect contentArea)
 	{
 		if (NeedsDisplay) {
 			if (SuperView != null) {
-				Clear (BoundsToScreen (Bounds));
+				Clear (BoundsToScreen (contentArea));
```
Comment (BDisp) [inline Terminal.Gui/View/ViewDrawing.cs:457]: `contentArea` is the same as `Bounds` but only screen relative, right?

File: Terminal.Gui/View/Layout/ViewLayout.cs [hunk 2]
```diff
@@ -191,15 +191,15 @@ void ThicknessChangedHandler (object sender, EventArgs e)
 			Margin.ThicknessChanged -= ThicknessChangedHandler;
 			Margin.Dispose ();
 		}
-		Margin = new Frame () { Id = "Margin", Thickness = new Thickness (0) };
+		Margin = new Frame () { Thickness = new Thickness (0) };
```
Comment (BDisp) [inline Terminal.Gui/View/Layout/ViewLayout.cs:194]: In the future if `Margin` has a different behavior than `Frame` then you will to sub-class it.

File: Terminal.Gui/View/Layout/ViewLayout.cs [hunk 3]
```diff
@@ -209,7 +209,7 @@ void ThicknessChangedHandler (object sender, EventArgs e)
 			Padding.ThicknessChanged -= ThicknessChangedHandler;
 			Padding.Dispose ();
 		}
-		Padding = new Frame () { Id = "Padding", Thickness = new Thickness (0) };
+		Padding = new Frame () { Thickness = new Thickness (0) };
```
Comment (BDisp) [inline Terminal.Gui/View/Layout/ViewLayout.cs:212]: In the future if `Padding` has a different behavior than `Frame` then you will to sub-class it.

File: Terminal.Gui/View/ViewDrawing.cs (hunk 1 above)
Comment (tig) [reply Terminal.Gui/View/ViewDrawing.cs:457]: Yes. I was going to remove that param from OnDrawContent but now I'm glad i never did. 

You'll see why soon...

File: Terminal.Gui/View/Layout/ViewLayout.cs (hunk 2 above)
Comment (tig) [reply Terminal.Gui/View/Layout/ViewLayout.cs:194]: Yessir. 

I already have one difference: Margin uses SuperView color scheme. Border and padding uses View.

**Comment by tig** [comment]
I started in on adding subviews to border. It's complicated so I'm not going to do it in this pr. I'll neaten this one up a bit more and merge it and do another for that.

**Comment by BDisp** [comment]
> I started in on adding subviews to border. It's complicated so I'm not going to do it in this pr. I'll neaten this one up a bit more and merge it and do another for that.

Ok, I also trying to use the `Padding` of the `TabView` to handle the tan rows and it isn't really easy because all the login for handles with views and sub-views are in View class itself. But I'll continue trying.
Why not change the `Frame` class name to `Adornment`. The sub-views that are added to them are still `View` sub-classes. Thus will not make confusing with the `Rect Frame`.

**Comment by tig** [comment]
On the padding thing: it'd be great if you opened an Issue on that. I'd love to dig into it with you with focus (and a simple example, not TabView). 

On name: I think you know I've been wishy-washy on Frame vs Adornment as the name. Your comment is helpful. I'll strongly consider it!

**Comment by BDisp** [comment]
> On the padding thing: it'd be great if you opened an Issue on that. I'd love to dig into it with you with focus (and a simple example, not TabView).
> 

I was trying in this PR #2980 but didn't commit/push nothing addressed with `Padding`. So, you don't want I continue on here?

> On name: I think you know I've been wishy-washy on Frame vs Adornment as the name. Your comment is helpful. I'll strongly consider it!

Ok at least I did you reflect about it 😃

**Comment by tig** [comment]
> > On the padding thing: it'd be great if you opened an Issue on that. I'd love to dig into it with you with focus (and a simple example, not TabView).
> 
> I was trying in this PR #2980 but didn't commit/push nothing addressed with `Padding`. So, you don't want I continue on here?

When working on architecture, I'm a fan of starting with just unit tests and a very simple Scenario vs. a major change like what needs to happen to TabView. TabView is a pretty complex beast, so yes.

**Comment by BDisp** [comment]
> > On name: I think you know I've been wishy-washy on Frame vs Adornment as the name. Your comment is helpful. I'll strongly consider it!
> 
> Ok at least I did you reflect about it 😃

I really appreciated for you have reconsider to change from `Frame` to `Adornment`. Thanks.

**Comment by BDisp** [comment]
@tig how about rename the `Bounds` to `ContentArea` as it's used on some of the `View` methods?

**Comment by tig** [comment]
> @tig how about rename the `Bounds` to `ContentArea` as it's used on some of the `View` methods?

- https://github.com/gui-cs/Terminal.Gui/issues/3169

**Pull request merged**